### PR TITLE
fix(apple): updating apple migration event based on fxa payloads

### PR DESCRIPTION
**Pull request by bassrock**
## Goal

Update the Pocket payload generation to be based on the schema that FxA is sending to us to migrate apple users.

```json
{
  "aud": "secret",
  "events": {
    "https://schemas.accounts.firefox.com/event/apple-user-migration": {
      "fxaEmail": "secret@email.com",
      "appleEmail": "secret@email.com",
      "transferSub": "secretId",
      "success": true,
      "uid": "secretUID"
    }
  },
  "iat": 1688074588.229,
  "jti": "id",
  "sub": "secretUID",
  "iss": "issuer"
}
```

## I'd love feedback/perspectives on:
- 

## Implementation Decisions
- 

## Deployment steps
- [ ] Database migrations?
- [ ] Secrets?
- [ ] **FOR POCKET DEVELOPERS ONLY** - Post in [\#changelog](https://pocket.slack.com/archives/C0Q4UFMDZ):
> Write a #changelog message using our [best practices](https://docs.google.com/document/d/1oEt8Mtkp-6Xz9S2zaNX1EIXfQIEDN2JpY0diuPc1HZc/edit) if this PR (potentially) impacts users. Describe the 'why' and 'what'. @mention relevant teams. Link to this PR.

## References

JIRA ticket:
* Link to JIRA ticket

Issue:
* Link to GitHub issue

Documentation:
* Project doc

### CODE REVIEW

**Comment by vbudhram** [comment]
@bassrock This matches what FxA is sending. I included `appleEmail` since I wasn't sure if Pocket needed to do an reconcilation with it. We could remove it if you want.

**Review by Herraj — APPROVED** [review]

File: gateway_lambda/config.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ const config = {
         EVENT.PROFILE_UPDATE,
       'https://schemas.accounts.firefox.com/event/delete-user':
         EVENT.USER_DELETE,
-      'https://schemas.accounts.firefox.com/event/apple-migration':
+      'https://schemas.accounts.firefox.com/event/apple-user-migration':
```
Comment (efixler) [inline gateway_lambda/config.ts:31]: Neither of these urls are accessible AFAICT. How do I know which one is right?

File: gateway_lambda/index.ts [hunk 2]
```diff
@@ -51,8 +51,8 @@ export function generateEvents(data: FxaPayload): SqsEvent[] {
       user_id: userId,
       event: allowedEvents[event],
       timestamp: Math.round(new Date().getTime() / 1000),
-      user_email: events[event].email,
-      transfer_sub: events[event].transfer_sub ?? null,
+      user_email: events[event].fxaEmail ?? events[event].email, //first see if we have an FxA email in an apple migration event, then fallback to email
+      transfer_sub: events[event].transferSub ?? null,
```
Comment (efixler) [inline gateway_lambda/index.ts:55]: What does transferSub mean?

**Comment by efixler** [comment]
@bassrock Is the expectation here that I would test this by checkout out the repo and running the unit test?

File: gateway_lambda/config.ts (hunk 1 above)
Comment (bassrock) [reply gateway_lambda/config.ts:31]: @efixler nope, these are schemea ids provided by fxa.

File: gateway_lambda/index.ts (hunk 2 above)
Comment (bassrock) [reply gateway_lambda/index.ts:55]: @efixler it is an identifier that apple generates to transfer users across apple teams. https://developer.apple.com/documentation/sign_in_with_apple/transferring_your_apps_and_users_to_another_team

**Comment by bassrock** [comment]
@efixler yea thats really the only way to test this. Its a close to do it live and line up schemas scenario.

**Comment by bassrock** [comment]
> @bassrock Is the expectation here that I would test this by checkout out the repo and running the unit test?

Which the CI also does.

File: gateway_lambda/config.ts (hunk 1 above)
Comment (efixler) [reply gateway_lambda/config.ts:31]: Ok @bassrock sorry if this is a dumb question, but don't these urls need to be be accessible?

Comment (bassrock) [reply gateway_lambda/config.ts:31]: No it doesn't they use them as an ID in the payload they send us. 

We should set up time and i can walk you through it. This repo is purely for web hooks https://mozilla.github.io/ecosystem-platform/relying-parties/tutorials/integration-with-fxa#webhook-events

**Review by efixler — APPROVED** [review]

**Pull request merged**